test/extended/builds/hooks.go

```diff
@@ -0,0 +1,85 @@
+package builds
+
+import (
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"time"
+)
+
+var _ = g.Describe("[builds][Slow] testing build configuration hooks", func() {
+	defer g.GinkgoRecover()
+	var (
+		buildFixture = exutil.FixturePath("testdata", "test-build-postcommit.json")
+		oc           = exutil.NewCLI("cli-test-hooks", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+		oc.Run("create").Args("-f", buildFixture).Execute()
+
+		g.By("waiting for istag to initialize")
+		wait.Poll(time.Second, 60*time.Second, func() (bool, error) {
+			err := oc.Run("get").Args("istag", "busybox:1").Execute()
+			if err == nil {
+				return true, nil
+			}
+			return false, nil
+		})
+
+	})
+
+	g.Describe("testing postCommit hook", func() {
+
+		g.It("successful postCommit script with args", func() {
+			err := oc.Run("patch").Args("bc/busybox", "-p", "{\"spec\":{\"postCommit\":{\"script\":\"echo hello $1\",\"args\":[\"world\"],\"command\":null}}}").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			br, _ := exutil.StartBuildAndWait(oc, "busybox")
+			br.AssertSuccess()
+			o.Expect(br.Logs()).To(o.ContainSubstring("hello world"))
+		})
+
+		g.It("successful postCommit explicit command", func() {
+			err := oc.Run("patch").Args("bc/busybox", "-p", "{\"spec\":{\"postCommit\":{\"command\":[\"sh\",\"-c\"],\"args\":[\"echo explicit command\"],\"script\":\"\"}}}").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			br, _ := exutil.StartBuildAndWait(oc, "busybox")
+			br.AssertSuccess()
+			o.Expect(br.Logs()).To(o.ContainSubstring("explicit command"))
+		})
+
+		g.It("successful postCommit default entrypoint", func() {
+			err := oc.Run("patch").Args("bc/busybox", "-p", "{\"spec\":{\"postCommit\":{\"args\":[\"echo\",\"default entrypoint\"],\"command\":null,\"script\":\"\"}}}").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			br, _ := exutil.StartBuildAndWait(oc, "busybox")
+			br.AssertSuccess()
+			o.Expect(br.Logs()).To(o.ContainSubstring("default entrypoint"))
+		})
+
+		g.It("failing postCommit script", func() {
+			err := oc.Run("patch").Args("bc/busybox", "-p", "{\"spec\":{\"postCommit\":{\"script\":\"false\",\"command\":null}}}").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			br, _ := exutil.StartBuildAndWait(oc, "busybox")
+			br.AssertFailure()
```

Comment:
needs to confirm the failure was due to the post commit hook running and failing.